field_RANKSUFF_module.fypp

```diff
@@ -158,6 +162,7 @@ CONTAINS
     SELF%THREAD_BUFFER = .NOT. LLPERSISTENT
     CALL SELF%SET_STATUS (NHSTFRESH)
 
+    SELF%MAP_DEVPTR = INIT_MAP_DEVPTR
```

Comment:
This overrides the default (.FALSE.) value we set in field_basic_module.F90. I think we should only have one mechanism to set the default value of `SELF%MAP_DEVPTR`. I like the configurable `INIT_MAP_DEVPTR`, so this should also be applied to OWNER_INIT as well, and we should remove the default value in field_basic_module.F90.